test/unit/worker/startup.test.ts

```diff
@@ -0,0 +1,185 @@
+		await enqueuePush(event.payload, process.env.ATLASSIAN_URL || "")
+	}
+
+	const mockGithubAccessToken = () => {
+		githubNock
+			.post("/app/installations/1234/access_tokens")
+			.optionally()
+			.reply(200, {
+				token: "token",
+				expires_at: new Date().getTime()
+			});
+	}
+
+	function mockGitHubAndJiraResponsesForRequestProcessing() {
+
+		mockGithubAccessToken();
+
+		githubNock.get("/repos/test-repo-owner/test-repo-name/commits/commit-no-username")
+			.reply(200, require("../../fixtures/push-non-merge-commit"));
+
+		// flag property should not be present
+		jiraNock.post("/rest/devinfo/0.10/bulk", {
+			preventTransitions: false,
+			repositories: [
+				{
+					name: "test-repo-name",
+					url: "test-repo-url",
+					id: "test-repo-id",
+					commits: [
+						{
+							hash: "commit-no-username",
+							message: "[TEST-123] Test commit.",
+							author: {
+								avatar: "https://github.com/users/undefined.png",
+								name: "test-commit-name",
+								email: "test-email@example.com",
+								url: "https://github.com/users/undefined",
+							},
+							authorTimestamp: "test-commit-date",
+							displayId: "commit",
+							fileCount: 3,
+							files: [
+								{
+									path: "test-modified",
+									changeType: "MODIFIED",
+									linesAdded: 10,
+									linesRemoved: 2,
+									url: "https://github.com/octocat/Hello-World/blob/7ca483543807a51b6079e54ac4cc392bc29ae284/test-modified"
+								},
+								{
+									path: "test-added",
+									changeType: "ADDED",
+									linesAdded: 4,
+									linesRemoved: 0,
+									url: "https://github.com/octocat/Hello-World/blob/7ca483543807a51b6079e54ac4cc392bc29ae284/test-added"
+								},
+								{
+									path: "test-removal",
+									changeType: "DELETED",
+									linesAdded: 0,
+									linesRemoved: 4,
+									url: "https://github.com/octocat/Hello-World/blob/7ca483543807a51b6079e54ac4cc392bc29ae284/test-removal"
+								}
+							],
+							id: "commit-no-username",
+							issueKeys: ["TEST-123"],
+							url: "https://github.com/octokit/Hello-World/commit/commit-no-username",
+							updateSequenceId: 12345678
+						}
+					],
+					updateSequenceId: 12345678
+				}
+			],
+			properties: {installationId: 1234}
+		}).reply(200);
+	}
+
+	it("should normally process push events sync after restart", async () => {
+
+		mockGitHubAndJiraResponsesForRequestProcessing();
+
+		await stop();
+
+		await start();
+
+		await enqueueTestPushEvent();
+
+		await waitUntil( async () => {
+			// eslint-disable-next-line jest/no-standalone-expect
+			expect(githubNock.pendingMocks()).toEqual([]);
+			// eslint-disable-next-line jest/no-standalone-expect
+			expect(jiraNock.pendingMocks()).toEqual([]);
+		})
+	});
+
+	it("should not process push events when stopped", async () => {
+
+		mockGitHubAndJiraResponsesForRequestProcessing();
+
+		//await stop();
+
+		await enqueueTestPushEvent();
+
+		delay(1000)
```

Comment:
There is no really better way to test it. We need to make sure that we NOT processing messages from the queue, which is tricky to do without waiting, hence we don't know when exactly worker is going to read.

Should be OK for now

Reply:
You're missing the await on this.

Reply:
The [jest recommended way is to use fake timers and move the clock forward](https://jestjs.io/docs/timer-mocks) so that your tests doesn't actually have to wait.